phoenix-core/src/main/java/org/apache/phoenix/log/QueryLogger.java

```diff
@@ -47,12 +48,14 @@ private QueryLogger(PhoenixConnection connection) {
         this.queryId = UUID.randomUUID().toString();
         this.queryDisruptor = connection.getQueryServices().getQueryDisruptor();
         logLevel = connection.getLogLevel();
+        auditLogLevel = connection.getAuditLogLevel();
```

Comment:
We're only using `logLevel` to determine when we should instantiate a real `QueryLogger` instance in `getInstance(PhoenixConnection, boolean)`.

Is the expectation that a user could have audit logging without "normal" logging enabled? Right now, it looks like you have to have at least some "normal" logging enabled to get the audit logging.

I think we would want these to be exclusive (if either normal or audit logging is enabled, we instantiate a real QueryLogger).

Reply:
(can't comment on this line in the code, so putting it here)

We also only get a real QueryLogger when `getLogSamplingRate()` matches the condition inside `getInstance(...)`. Is it OK to have just one sampling rate for both normal and audit logging?

Reply:
Great catch Josh!

We would like the QueryLogger work separately from the newly created AuditQueryLogger.
We should log DML and DDL operations when QueryServices.AUDIT_LOG_LEVEL is greater than INFO even if normal logging is disabled. (the default is OFF)
We can have both enabled and if we want to have logging for select queries and for DML + DDL queries, we should.

About the LogSamplingRate
In the latest change I separated AuditQueryLogger and LogSamplingRate has no effect on it, I think If AUDIT_LOG_LEVEL is enabled we should log every query.

Reply:
> We can have both enabled and if we want to have logging for select queries and for DML + DDL queries, we should.

Great! That sounds like the correct approach.

> In the latest change I separated AuditQueryLogger and LogSamplingRate has no effect on it, I think If AUDIT_LOG_LEVEL is enabled we should log every query.

Ok, I think that is fine. Let's make sure we circle back around an update Phoenix documentation for this new feature.